core/src/main/java/org/elasticsearch/action/fieldcaps/FieldCapabilities.java

```diff
@@ -227,7 +227,7 @@ void add(String index, boolean search, boolean agg) {
 
         FieldCapabilities build(boolean withIndices) {
             final String[] indices;
-            Collections.sort(indiceList, Comparator.comparing(o -> o.name));
+            Collections.sort(indiceList, Comparator.comparing((IndexCaps o) -> o.name));
```

Comment:
Please add a comment why this is so lest it get inadvertently removed before Eclipse can handle the truth.